hcontext/requestid.go

```diff
@@ -22,12 +22,14 @@ var headersToSearch = []string{
 	"Request-ID", "X-Request-ID",
 }
 
-// FromRequest fetches the given request's request ID if it has one,
-// and returns a new random request ID if it does not.
+// FromRequest fetches the given request's request ID from the Headers.
+// If one is found, it appends a new request ID and sets the comma separated value as the header.
+// If one is not found, it sets a new request ID as the header.
 func FromRequest(r *http.Request) (id string, ok bool) {
 	for _, try := range headersToSearch {
 		if id = r.Header.Get(try); id != "" {
-			return id, true
+			newRequestID := uuid.New().String() + "," + id
```

Comment:
@danp yes, you are right, this is prepending a new ID so if you called it multiple times it would keep prepending new UUIDs. This is not super problematic in my view; it seemed less confusing to add this functionality to a current function than to add yet another request_id-related function.

List of existing Functions:

`RequestIdFromContext`, which retrieves request_id from context https://github.com/heroku/x/blob/79265fb93afba37611483937cf898a96af8cea2f/hcontext/requestid.go#L46

`WithRequestID`, adds request_id to Context https://github.com/heroku/x/blob/79265fb93afba37611483937cf898a96af8cea2f/hcontext/requestid.go#L41

`Get`, which retrieves request_ids from Headers https://github.com/heroku/x/blob/79265fb93afba37611483937cf898a96af8cea2f/requestid/requestid.go#L11

And then this one, `FromRequest`, which fetches the request_id from the headers and if there is not one returns a new UUID. 

It seems like if someone wanted to just retrieve the request_id, they could call `Get` and not worry about more request_ids being added.